### PR TITLE
Bug/gen empty output

### PR DESCRIPTION
**Pull request by jonathangomesselman**
* [x] I have added tests to `tests` to cover my changes.
* [ ] I have updated `docs/`, if necessary.
* [ ] I have updated the `README.md`, if necessary.

***What existing issue does this pull request close?***

Shortcut ticket: [link](https://app.shortcut.com/galileo/epic/5806?cf_workflow=500000005&ct_workflow=all&group_by=none&vc_group_by=day)

***Bug Description***
DQ logging fails when handling sequences of length one. This can happen for example in the case of an empty string with T5 where T5 represents the empty string as `input_ids = [1]` (i.e. just the `[EOS]` token). So e.g. during generation the model may immediately generate the `[EOS]` token leading, leading to a single token sequence.

***Root cause analysis***
This bug is a result of using the `.squeeze()` function without properly specifying dimensions. We use `.squeeze()` to remove singleton e.g. batch dimensions; however, when we have `seq_len=1`, `.squeeze()` will also remove this sequence dimension. See this example:
```
gen_ids = gen_ids[..., 1:] # Shape - (1, seq_len)
....
gen_ids = gen_ids.squeeze().cpu().numpy() # [seq_len]
```
If `seq_len = 1` then `gen_ids` will have  `shape = ()` rather than `(seq_len)`, since `squeeze()` removes all singleton dimensions. 

The fix that we add is that whenever we use squeeze we specify which dimension we specifically are squeezing! E.g.
```
gen_ids = gen_ids[..., 1:] # Shape - (1, seq_len)
....
# Squeeze the batch dimension
gen_ids = gen_ids.squeeze(0).cpu().numpy() # [seq_len]
```

### CODE REVIEW

File: tests/utils/test_seq2seq_utils.py [hunk 1]
```diff
@@ -99,6 +99,55 @@ class FakeOutput:
...
+    # Mock the model
+    mock_model = mock.MagicMock()
+    # Add a fake <pad> token to the generated ids
+    mock_model.generate.return_value = torch.tensor([[0, 1]])
+
+    # Mock the model forward function to return random logits
+    # for a single batch element
+    @dataclass
+    class FakeOutput:
+        logits: torch.tensor
```
Comment (elboy3) [inline tests/utils/test_seq2seq_utils.py:130]: you can use a mock for this return value too! 

`mock.MagicMock(logits=torch.rand((1, 1, 20)))`

**Review by elboy3 — APPROVED** [review]
one suggestion. sick!

File: tests/utils/test_seq2seq_utils.py (hunk 1 above)
Comment (jonathangomesselman) [reply tests/utils/test_seq2seq_utils.py:130]: Oh heck yeah!

**Comment by bogdan-galileo** [comment]
This also takes care of https://app.shortcut.com/galileo/story/7869/dq-logging-breaks-if-a-sample-has-empty-target. Great!
